openstack/networking/v2/ports/requests.go

```diff
@@ -150,14 +151,15 @@ type UpdateOptsBuilder interface {
 
 // UpdateOpts represents the attributes used when updating an existing port.
 type UpdateOpts struct {
-	Name                *string        `json:"name,omitempty"`
-	Description         *string        `json:"description,omitempty"`
-	AdminStateUp        *bool          `json:"admin_state_up,omitempty"`
-	FixedIPs            interface{}    `json:"fixed_ips,omitempty"`
-	DeviceID            *string        `json:"device_id,omitempty"`
-	DeviceOwner         *string        `json:"device_owner,omitempty"`
-	SecurityGroups      *[]string      `json:"security_groups,omitempty"`
-	AllowedAddressPairs *[]AddressPair `json:"allowed_address_pairs,omitempty"`
+	Name                *string           `json:"name,omitempty"`
+	Description         *string           `json:"description,omitempty"`
+	AdminStateUp        *bool             `json:"admin_state_up,omitempty"`
+	FixedIPs            interface{}       `json:"fixed_ips,omitempty"`
+	DeviceID            *string           `json:"device_id,omitempty"`
+	DeviceOwner         *string           `json:"device_owner,omitempty"`
+	SecurityGroups      *[]string         `json:"security_groups,omitempty"`
+	AllowedAddressPairs *[]AddressPair    `json:"allowed_address_pairs,omitempty"`
+	ValueSpecs          map[string]string `json:"value_specs,omitempty"`
```

Comment:
I know it's horrible, but could you please change this field to be a pointer?
Before your change, `UpdateOpts` is [comparable](https://go.dev/ref/spec#Comparison_operators). Having it as a pointer would keep it comparable, and thus would not break compilation of any consumer code.

Reply:
Good point, done!

Reply:
Argh... messed up the branches. go mod does not like branches with `/` in the name, and I'm using the branch here for a PR in CAPO (https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1452), which means I had to change the branch name.
I will create a new PR with the correct branch name.

Reply:
Ok, thanks! I am closing this one to avoid confusion.

Reply:
The new PR is https://github.com/gophercloud/gophercloud/pull/2537. It has the same content (with the fixed the pointer). Sorry for the mess.

Ref for go issue with slashes in branch names: https://github.com/golang/go/issues/32955#issuecomment-511860346

Reply:
Good to know!